community/kernel/src/main/java/org/neo4j/unsafe/impl/batchimport/store/BatchingNeoStores.java

```diff
@@ -73,90 +77,147 @@
  */
 public class BatchingNeoStores implements AutoCloseable, MemoryStatsVisitor.Visitable
 {
+    private static final String TEMP_NEOSTORE_NAME = "temp." + DEFAULT_NAME;
+
     private final FileSystemAbstraction fileSystem;
-    private final BatchingPropertyKeyTokenRepository propertyKeyRepository;
-    private final BatchingLabelTokenRepository labelRepository;
-    private final BatchingRelationshipTypeTokenRepository relationshipTypeRepository;
     private final LogProvider logProvider;
     private final File storeDir;
     private final Config neo4jConfig;
     private final Configuration importConfiguration;
     private final PageCache pageCache;
-    private final NeoStores neoStores;
-    private final LifeSupport life = new LifeSupport();
-    private final LabelScanStore labelScanStore;
     private final IoTracer ioTracer;
     private final RecordFormats recordFormats;
+    private final AdditionalInitialIds initialIds;
+    private final boolean externalPageCache;
 
     // Some stores are considered temporary during the import and will be reordered/restructured
     // into the main store. These temporary stores will live here
-    private final NeoStores temporaryNeoStores;
-    private final boolean externalPageCache;
+    private NeoStores neoStores;
+    private NeoStores temporaryNeoStores;
+    private BatchingPropertyKeyTokenRepository propertyKeyRepository;
+    private BatchingLabelTokenRepository labelRepository;
+    private BatchingRelationshipTypeTokenRepository relationshipTypeRepository;
+    private LifeSupport life = new LifeSupport();
+    private LabelScanStore labelScanStore;
     private PageCacheFlusher flusher;
 
+    private boolean successful;
+
     private BatchingNeoStores( FileSystemAbstraction fileSystem, PageCache pageCache, File storeDir,
             RecordFormats recordFormats, Config neo4jConfig, Configuration importConfiguration, LogService logService,
             AdditionalInitialIds initialIds, boolean externalPageCache, IoTracer ioTracer )
     {
         this.fileSystem = fileSystem;
         this.recordFormats = recordFormats;
         this.importConfiguration = importConfiguration;
+        this.initialIds = initialIds;
         this.logProvider = logService.getInternalLogProvider();
         this.storeDir = storeDir;
         this.neo4jConfig = neo4jConfig;
         this.pageCache = pageCache;
         this.ioTracer = ioTracer;
         this.externalPageCache = externalPageCache;
-        this.neoStores = newStoreFactory( DEFAULT_NAME ).openAllNeoStores( true );
-        if ( alreadyContainsData( neoStores ) )
+    }
+
+    private boolean databaseExistsAndContainsData( PageCache pageCache, File storeDir )
+    {
+        File metaDataFile = new File( storeDir, StoreType.META_DATA.getStoreFile().fileName( StoreFileType.STORE ) );
+        try ( PagedFile pagedFile = pageCache.map( metaDataFile, pageCache.pageSize(), StandardOpenOption.READ ); )
```

Comment:
it should be enough to check if META_DATA store file is present right without all this mapping dance?

Reply:
No, apparently it's not. We do something strange in store migration where some files are present at the time of starting a migration.